The job **`x86_64-gnu-llvm-20`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/22529495529/job/65266725859) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/65266725859) [(plain)](https://github.com/rust-lang/rust/commit/30c2d6bbffbced9e50b392a2fc7091867a711e98/checks/65266725859/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
/dev/sda15      105M  6.2M   99M   6% /boot/efi
tmpfs           1.6G   12K  1.6G   1% /run/user/1001
================================================================================

Sufficient disk space available (94917592KB >= 52428800KB). Skipping cleanup.
##[group]Run echo "[CI_PR_NUMBER=$num]"
echo "[CI_PR_NUMBER=$num]"
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
---
set -ex

# Run a subset of tests. Used to run tests in parallel in multiple jobs.

# When this job partition is run as part of PR CI, skip tidy to allow revealing more failures. The
# dedicated `tidy` job failing won't block other PR CI jobs from completing, and so tidy failures
# shouldn't inhibit revealing other failures in PR CI jobs.
if [ "$PR_CI_JOB" == "1" ]; then
  echo "PR_CI_JOB set; skipping tidy"
  SKIP_TIDY="--skip tidy"
fi

../x.py --stage 2 test \
  ${SKIP_TIDY:+$SKIP_TIDY} \
  --skip compiler \
  --skip src
#!/bin/bash

set -ex

# Run a subset of tests. Used to run tests in parallel in multiple jobs.

# When this job partition is run as part of PR CI, skip tidy to allow revealing more failures. The
# dedicated `tidy` job failing won't block other PR CI jobs from completing, and so tidy failures
# shouldn't inhibit revealing other failures in PR CI jobs.
if [ "$PR_CI_JOB" == "1" ]; then
  echo "PR_CI_JOB set; skipping tidy"
  SKIP_TIDY="--skip tidy"
fi

../x.py --stage 2 test \
  ${SKIP_TIDY:+$SKIP_TIDY} \
  --skip tests \
  --skip coverage-map \
  --skip coverage-run \
  --skip library \
  --skip tidyselftest
---
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/i686-unknown-linux-gnu/src/gcc/diagnostics/context.cc: In function ‘void fancy_abort(const char*, int, const char*)’:
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/i686-unknown-linux-gnu/src/gcc/diagnostics/context.cc:1796:15: warning: format not a string literal and no format arguments [-Wformat-security]
 1796 |       fnotice (stderr, diagnostics::get_text_for_kind (diagnostics::kind::ice));
      |       ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/i686-unknown-linux-gnu/src/gcc/diagnostics/digraphs-to-dot-from-cfg.cc: In member function ‘virtual void diagnostics::digraphs::to_dot::converter_from_cfg::add_any_node_attrs(const diagnostics::digraphs::to_dot::digraph_node&, dot::node_stmt&)’:
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/i686-unknown-linux-gnu/src/gcc/diagnostics/digraphs-to-dot-from-cfg.cc:173:36: warning: spurious leading punctuation sequence ‘<’ in format [-Wformat-diag]
  173 |                   pp_printf (&pp, "<bb %li>:", bb_index);
      |                                    ^
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/i686-unknown-linux-gnu/src/gcc/diagnostics/digraphs-to-dot-from-cfg.cc:173:43: warning: spurious trailing punctuation sequence ‘>:’ in format [-Wformat-diag]
  173 |                   pp_printf (&pp, "<bb %li>:", bb_index);
      |                                           ^~
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/i686-unknown-linux-gnu/src/gcc/diagnostics/digraphs-to-dot-from-cfg.cc: In member function ‘virtual void diagnostics::digraphs::to_dot::converter_from_cfg::add_any_edge_attrs(const diagnostics::digraphs::to_dot::digraph_edge&, dot::edge_stmt&)’:
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/i686-unknown-linux-gnu/src/gcc/diagnostics/digraphs-to-dot-from-cfg.cc:246:26: warning: spurious leading punctuation sequence ‘[’ in format [-Wformat-diag]
  246 |         pp_printf (&pp, "[%li%%]", probability_pc);
      |                          ^
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/i686-unknown-linux-gnu/src/gcc/diagnostics/sarif-sink.cc: In member function ‘void diagnostics::sarif_socket_sink::send_rpc_notification(const json::object&)’:
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/i686-unknown-linux-gnu/src/gcc/diagnostics/sarif-sink.cc:4282:11: warning: ignoring return value of ‘ssize_t write(int, const void*, size_t)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
 4282 |     write (m_fd, buf, output_size);
      |     ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/i686-unknown-linux-gnu/src/gcc/text-art/style.cc: In member function ‘void text_art::style::color::print_sgr(pretty_printer*, bool, bool&) const’:
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/i686-unknown-linux-gnu/src/gcc/text-art/style.cc:149:25: warning: spurious leading punctuation sequence ‘;’ in format [-Wformat-diag]
  149 |         pp_printf (pp, ";5;%i", (int)u.m_8bit);
      |                         ^
---
test [ui] tests/ui/abi/abi-sysv64-arg-passing.rs ... ok
test [ui] tests/ui/abi/abi-typo-unstable.rs#feature_disabled ... ok
test [ui] tests/ui/abi/abi-typo-unstable.rs#feature_enabled ... ok
test [ui] tests/ui/abi/anon-extern-mod.rs ... ok
test [ui] tests/ui/abi/avr-sram.rs#disable_sram ... ok
test [ui] tests/ui/abi/avr-sram.rs#has_sram ... ok
test [ui] tests/ui/abi/avr-sram.rs#no_sram ... ok
test [ui] tests/ui/abi/arm-unadjusted-intrinsic.rs#aarch64 ... ok
test [ui] tests/ui/abi/arm-unadjusted-intrinsic.rs#arm ... ok
test [ui] tests/ui/abi/c-stack-as-value.rs ... ok
test [ui] tests/ui/abi/c-stack-returning-int64.rs ... ok
test [ui] tests/ui/abi/bad-custom.rs ... ok
---
test [ui] tests/ui/asm/aarch64/type-f16.rs ... ignored, only executed when the architecture is aarch64
test [ui] tests/ui/array-slice-vec/vector-no-ann.rs ... ok
test [ui] tests/ui/array-slice-vec/vector-slice-matching-8498.rs ... ok
test [ui] tests/ui/asm/aarch64/arm64ec-sve.rs ... ok
test [ui] tests/ui/asm/aarch64v8r.rs#hf ... ok
test [ui] tests/ui/asm/aarch64v8r.rs#r82 ... ok
test [ui] tests/ui/asm/aarch64v8r.rs#sf ... ok
test [ui] tests/ui/asm/asm-with-nested-closure.rs ... ok
test [ui] tests/ui/asm/binary_asm_labels.rs ... ignored, only executed when the architecture is x86_64
test [ui] tests/ui/asm/binary_asm_labels_allowed.rs ... ignored, only executed when the architecture is aarch64
test [ui] tests/ui/asm/cfg-parse-error.rs ... ok
test [ui] tests/ui/asm/cfg.rs#reva ... ignored, only executed when the architecture is x86_64
---
test [ui] tests/ui/const-generics/ogca/basic-fail.rs ... ok
test [ui] tests/ui/const-generics/occurs-check/unused-substs-4.rs ... ok
test [ui] tests/ui/const-generics/ogca/basic.rs ... ok
test [ui] tests/ui/const-generics/occurs-check/unused-substs-5.rs ... ok
test [ui] tests/ui/const-generics/ogca/rhs-but-not-root.rs ... ok
test [ui] tests/ui/const-generics/ogca/generic-param-rhs.rs ... ok
test [ui] tests/ui/const-generics/ogca/coherence-ambiguous.rs ... ok
test [ui] tests/ui/const-generics/outer-lifetime-in-const-generic-default.rs ... ok
test [ui] tests/ui/const-generics/opaque_types.rs ... ok
test [ui] tests/ui/const-generics/overlapping_impls.rs ... ok
test [ui] tests/ui/const-generics/opaque_types2.rs ... ok
test [ui] tests/ui/const-generics/params-in-ct-in-ty-param-lazy-norm.rs#min ... ok
---
test [ui] tests/ui/extern/issue-64655-extern-rust-must-allow-unwind.rs#thin1 ... ok
test [ui] tests/ui/extern/issue-64655-extern-rust-must-allow-unwind.rs#thin2 ... ok
test [ui] tests/ui/extern/issue-64655-extern-rust-must-allow-unwind.rs#thin3 ... ok
test [ui] tests/ui/extern/issue-95829.rs ... ok
test [ui] tests/ui/extern/lgamma-linkage.rs ... ok
test [ui] tests/ui/extern/not-in-block.rs ... ok
test [ui] tests/ui/extern/unsized-extern-derefmove.rs ... ok
test [ui] tests/ui/extern/windows-tcb-trash-13259.rs ... ok
test [ui] tests/ui/feature-gates/allow-features.rs ... ok
test [ui] tests/ui/extern/no-mangle-associated-fn.rs ... ok
---
test [ui] tests/ui/feature-gates/feature-gate-macro-attr.rs ... ok
test [ui] tests/ui/feature-gates/feature-gate-macro-metavar-expr-concat.rs ... ok
test [ui] tests/ui/feature-gates/feature-gate-marker_trait_attr.rs ... ok
test [ui] tests/ui/feature-gates/feature-gate-may-dangle.rs ... ok
test [ui] tests/ui/feature-gates/feature-gate-mgca-type-const-syntax.rs ... ok
test [ui] tests/ui/feature-gates/feature-gate-min-generic-const-args.rs ... ok
test [ui] tests/ui/feature-gates/feature-gate-movrs_target_feature.rs ... ignored, only executed when the architecture is x86_64
test [ui] tests/ui/feature-gates/feature-gate-min_const_fn.rs ... ok
test [ui] tests/ui/feature-gates/feature-gate-more-maybe-bounds.rs ... ok
test [ui] tests/ui/feature-gates/feature-gate-naked_functions_rustic_abi.rs ... ignored, only executed when the architecture is x86_64
---
test [ui] tests/ui/imports/ambiguous-7.rs ... ok
test [ui] tests/ui/imports/ambiguous-import-visibility-module.rs ... ok
test [ui] tests/ui/imports/ambiguous-8.rs ... ok
test [ui] tests/ui/imports/ambiguous-glob-vs-expanded-extern.rs ... ok
test [ui] tests/ui/imports/ambiguous-panic-glob-vs-multiouter.rs ... ok
test [ui] tests/ui/imports/ambiguous-panic-globvsglob.rs ... ok
test [ui] tests/ui/imports/ambiguous-panic-no-implicit-prelude.rs ... ok
test [ui] tests/ui/imports/ambiguous-panic-non-prelude-core-glob.rs ... ok
test [ui] tests/ui/imports/ambiguous-panic-non-prelude-std-glob.rs ... ok
test [ui] tests/ui/imports/ambiguous-panic-pick-core.rs ... ok
test [ui] tests/ui/imports/ambiguous-panic-pick-std.rs ... ok
---
test [ui] tests/ui/layout/null-pointer-optimization.rs ... ok
test [ui] tests/ui/layout/randomize.rs#normal ... ok
test [ui] tests/ui/layout/randomize.rs#randomize-layout ... ok
test [ui] tests/ui/layout/post-mono-layout-cycle.rs ... ok
test [ui] tests/ui/layout/rigid-alias-due-to-broken-impl.rs ... ok
test [ui] tests/ui/layout/rigid-alias-no-params.rs ... ok
test [ui] tests/ui/layout/reprc-power-alignment.rs ... ok
test [ui] tests/ui/layout/struct.rs ... ok
test [ui] tests/ui/layout/rust-call-abi-not-a-tuple-ice-81974.rs ... ok
test [ui] tests/ui/layout/thaw-transmute-invalid-enum.rs ... ok
test [ui] tests/ui/layout/thaw-validate-invalid-enum.rs ... ok
---

4 LL | type const CONST: usize = 1_i32;
5    | ^^^^^^^^^^^^^^^^^^^^^^^ expected `usize`, found `i32`
6 
- error[E0080]: transmuting from 4-byte type to 8-byte type: `i32` -> `usize`
-   --> $DIR/type_const-mismatched-literal-suffix.rs:11:13
-    |
- LL |     const { CONST };
-    |             ^^^^^ evaluation of `main::{constant#0}` failed here
+ error: aborting due to 1 previous error
12 
- note: erroneous constant encountered
-   --> $DIR/type_const-mismatched-literal-suffix.rs:11:5
-    |
---
-    |
- LL |     const { CONST };
-    |     ^^^^^^^^^^^^^^^
-    |
-    = note: duplicate diagnostic emitted due to `-Z deduplicate-diagnostics=no`
- 
- error: aborting due to 2 previous errors
- 
- For more information about this error, try `rustc --explain E0080`.
30 
---
To only update this specific test, also pass `--test-args const-generics/mgca/type_const-mismatched-literal-suffix.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/const-generics/mgca/type_const-mismatched-literal-suffix.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=i686-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/const-generics/mgca/type_const-mismatched-literal-suffix" "-A" "unused" "-W" "unused_attributes" "-A" "internal_features" "-A" "unused_parens" "-A" "unused_braces" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/i686-unknown-linux-gnu/native/rust-test-helpers" "-Clinker=x86_64-linux-gnu-gcc" "--emit=link"
stdout: none
--- stderr -------------------------------
error: the constant `1` is not of type `usize`
##[error]  --> /checkout/tests/ui/const-generics/mgca/type_const-mismatched-literal-suffix.rs:6:1
   |
```

</details>

--- a/compiler/rustc_middle/src/ty/consts/lit.rs
+++ b/compiler/rustc_middle/src/ty/consts/lit.rs
@@ -1,4 +1,4 @@
-use rustc_ast::LitKind;
+use rustc_ast::{LitFloatType, LitIntType, LitKind};
 use rustc_hir;
 use rustc_macros::HashStable;
 
@@ -44,10 +44,17 @@ pub fn const_lit_matches_ty<'tcx>(
         {
             true
         }
-        (LitKind::Int(..), ty::Uint(_)) if !neg => true,
-        (LitKind::Int(..), ty::Int(_)) => true,
+        (LitKind::Int(_, LitIntType::Unsigned(lit_ty)), ty::Uint(expect_ty)) if !neg => {
+            lit_ty == *expect_ty
+        }
+        (LitKind::Int(_, LitIntType::Signed(lit_ty)), ty::Int(expect_ty)) => lit_ty == *expect_ty,
+        (LitKind::Int(_, LitIntType::Unsuffixed), ty::Uint(_)) if !neg => true,
+        (LitKind::Int(_, LitIntType::Unsuffixed), ty::Int(_)) => true,
         (LitKind::Bool(..), ty::Bool) => true,
-        (LitKind::Float(..), ty::Float(_)) => true,
+        (LitKind::Float(_, LitFloatType::Suffixed(lit_ty)), ty::Float(expect_ty)) => {
+            lit_ty == *expect_ty
+        }
+        (LitKind::Float(_, LitFloatType::Unsuffixed), ty::Float(_)) => true,
         (LitKind::Char(..), ty::Char) => true,
         (LitKind::Err(..), _) => true,
         _ => false,

--- a/tests/ui/const-generics/mgca/const-arg-mismatched-literal-suffix.rs
+++ b/tests/ui/const-generics/mgca/const-arg-mismatched-literal-suffix.rs
@@ -1,0 +1,10 @@
+#![feature(min_generic_const_args)]
+#![expect(incomplete_features)]
+
+type const CONST: usize = 1_i32;
+//~^ ERROR the constant `1` is not of type `usize`
+//~| NOTE expected `usize`, found `i32`
+
+fn main() {
+    const { CONST };
+}

--- a/tests/ui/const-generics/mgca/const-arg-mismatched-literal-suffix.stderr
+++ b/tests/ui/const-generics/mgca/const-arg-mismatched-literal-suffix.stderr
@@ -1,0 +1,8 @@
+error: the constant `1` is not of type `usize`
+  --> $DIR/const-arg-mismatched-literal-suffix.rs:4:1
+   |
+LL | type const CONST: usize = 1_i32;
+   | ^^^^^^^^^^^^^^^^^^^^^^^ expected `usize`, found `i32`
+
+error: aborting due to 1 previous error
+

--- a/tests/ui/const-generics/type-dependent/type-mismatch.full.stderr
+++ b/tests/ui/const-generics/type-dependent/type-mismatch.full.stderr
@@ -1,15 +1,3 @@
-error: the constant `1` is not of type `u8`
-  --> $DIR/type-mismatch.rs:8:27
-   |
-LL |     assert_eq!(R.method::<1u16>(), 1);
-   |                           ^^^^ expected `u8`, found `u16`
-   |
-note: required by a const generic parameter in `R::method`
-  --> $DIR/type-mismatch.rs:5:15
-   |
-LL |     fn method<const N: u8>(&self) -> u8 { N }
-   |               ^^^^^^^^^^^ required by this const generic parameter in `R::method`
-
 error[E0308]: mismatched types
   --> $DIR/type-mismatch.rs:8:27
    |
@@ -22,6 +10,6 @@ LL -     assert_eq!(R.method::<1u16>(), 1);
 LL +     assert_eq!(R.method::<1u8>(), 1);
    |
 
-error: aborting due to 2 previous errors
+error: aborting due to 1 previous error
 
 For more information about this error, try `rustc --explain E0308`.

--- a/tests/ui/const-generics/type-dependent/type-mismatch.min.stderr
+++ b/tests/ui/const-generics/type-dependent/type-mismatch.min.stderr
@@ -1,15 +1,3 @@
-error: the constant `1` is not of type `u8`
-  --> $DIR/type-mismatch.rs:8:27
-   |
-LL |     assert_eq!(R.method::<1u16>(), 1);
-   |                           ^^^^ expected `u8`, found `u16`
-   |
-note: required by a const generic parameter in `R::method`
-  --> $DIR/type-mismatch.rs:5:15
-   |
-LL |     fn method<const N: u8>(&self) -> u8 { N }
-   |               ^^^^^^^^^^^ required by this const generic parameter in `R::method`
-
 error[E0308]: mismatched types
   --> $DIR/type-mismatch.rs:8:27
    |
@@ -22,6 +10,6 @@ LL -     assert_eq!(R.method::<1u16>(), 1);
 LL +     assert_eq!(R.method::<1u8>(), 1);
    |
 
-error: aborting due to 2 previous errors
+error: aborting due to 1 previous error
 
 For more information about this error, try `rustc --explain E0308`.

--- a/tests/ui/const-generics/type-dependent/type-mismatch.rs
+++ b/tests/ui/const-generics/type-dependent/type-mismatch.rs
@@ -6,6 +6,5 @@ impl R {
 }
 fn main() {
     assert_eq!(R.method::<1u16>(), 1);
-    //~^ ERROR the constant `1` is not of type `u8`
-    //~| ERROR mismatched types
+    //~^ ERROR mismatched types
 }

--- a/tests/ui/consts/const-eval/array-len-mismatch-type.rs
+++ b/tests/ui/consts/const-eval/array-len-mismatch-type.rs
@@ -1,8 +1,5 @@
 //! Regression test for <https://github.com/rust-lang/rust/issues/133966>
 pub struct Data([[&'static str]; 5_i32]);
-//~^ ERROR the constant `5` is not of type `usize`
-//~| ERROR the size for values of type `[&'static str]` cannot be known at compilation time
-//~| ERROR mismatched types
+//~^ ERROR mismatched types
 const _: &'static Data = unsafe { &*(&[] as *const Data) };
-//~^ ERROR the type `[[&str]; 5]` has an unknown layout
 fn main() {}

--- a/tests/ui/consts/const-eval/array-len-mismatch-type.stderr
+++ b/tests/ui/consts/const-eval/array-len-mismatch-type.stderr
@@ -1,26 +1,3 @@
-error: the constant `5` is not of type `usize`
-  --> $DIR/array-len-mismatch-type.rs:2:17
-   |
-LL | pub struct Data([[&'static str]; 5_i32]);
-   |                 ^^^^^^^^^^^^^^^^^^^^^^^ expected `usize`, found `i32`
-   |
-   = note: the length of array `[[&'static str]; 5]` must be type `usize`
-
-error[E0277]: the size for values of type `[&'static str]` cannot be known at compilation time
-  --> $DIR/array-len-mismatch-type.rs:2:17
-   |
-LL | pub struct Data([[&'static str]; 5_i32]);
-   |                 ^^^^^^^^^^^^^^^^^^^^^^^ doesn't have a size known at compile-time
-   |
-   = help: the trait `Sized` is not implemented for `[&'static str]`
-   = note: slice and array elements must have `Sized` type
-
-error[E0080]: the type `[[&str]; 5]` has an unknown layout
-  --> $DIR/array-len-mismatch-type.rs:6:39
-   |
-LL | const _: &'static Data = unsafe { &*(&[] as *const Data) };
-   |                                       ^^ evaluation of `_` failed here
-
 error[E0308]: mismatched types
   --> $DIR/array-len-mismatch-type.rs:2:34
    |
@@ -33,7 +10,6 @@ LL - pub struct Data([[&'static str]; 5_i32]);
 LL + pub struct Data([[&'static str]; 5_usize]);
    |
 
-error: aborting due to 4 previous errors
+error: aborting due to 1 previous error
 
-Some errors have detailed explanations: E0080, E0277, E0308.
-For more information about an error, try `rustc --explain E0080`.
+For more information about this error, try `rustc --explain E0308`.

--- a/tests/ui/repeat-expr/repeat_count.rs
+++ b/tests/ui/repeat-expr/repeat_count.rs
@@ -27,10 +27,7 @@ fn main() {
     //~| NOTE expected `usize`, found `isize`
     //~| NOTE `-1_isize` cannot fit into type `usize`
     let h = [0; 4u8];
-    //~^ ERROR the constant `4` is not of type `usize`
-    //~| NOTE expected `usize`, found `u8`
-    //~| NOTE the length of array `[{integer}; 4]` must be type `usize`
-    //~| ERROR mismatched types
+    //~^ ERROR mismatched types
     //~| NOTE expected `usize`, found `u8`
     struct I {
         i: (),

--- a/tests/ui/repeat-expr/repeat_count.stderr
+++ b/tests/ui/repeat-expr/repeat_count.stderr
@@ -50,20 +50,6 @@ LL |     let g = [0_usize; -1_isize];
    |
    = note: `-1_isize` cannot fit into type `usize`
 
-error: the constant `4` is not of type `usize`
-  --> $DIR/repeat_count.rs:29:13
-   |
-LL |     let h = [0; 4u8];
-   |             ^^^^^^^^ expected `usize`, found `u8`
-   |
-   = note: the length of array `[{integer}; 4]` must be type `usize`
-
-error[E0308]: mismatched types
-  --> $DIR/repeat_count.rs:38:17
-   |
-LL |     let i = [0; I { i: () }];
-   |                 ^^^^^^^^^^^ expected `usize`, found `I`
-
 error[E0308]: mismatched types
   --> $DIR/repeat_count.rs:29:17
    |
@@ -76,7 +62,13 @@ LL -     let h = [0; 4u8];
 LL +     let h = [0; 4usize];
    |
 
-error: aborting due to 10 previous errors
+error[E0308]: mismatched types
+  --> $DIR/repeat_count.rs:35:17
+   |
+LL |     let i = [0; I { i: () }];
+   |                 ^^^^^^^^^^^ expected `usize`, found `I`
+
+error: aborting due to 9 previous errors
 
 Some errors have detailed explanations: E0308, E0435.
 For more information about an error, try `rustc --explain E0308`.

